File: server/Data/Entities/DetailedDeputado.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace server.Data.Entities
{
    public class DetailedDeputado
    {
        public int id { get; set; }
        public string uri { get; set; }
        public string nomeCivil { get; set; }
        public UltimoStatus ultimoStatus { get; set; }
        public string cpf { get; set; }
        public string sexo { get; set; }
        public string urlWebsite { get; set; }
        public string dataNascimento { get; set; }
        public string dataFalecimento { get; set; }
        public string ufNascimento { get; set; }
        public string municipioNascimento { get; set; }
        public string escolaridade { get; set; }
        public string email { get; set; }
        public List<Despesa> despesas { get; set; }
        public List<Frente> frentes { get; set; }
        public List<Orgao> orgaos { get; set; }
    }
    public class Gabinete
    {
        public int id { get; set; }
        public string nome { get; set; }
        public string predio { get; set; }
        public string sala { get; set; }
        public string andar { get; set; }
        public string telefone { get; set; }
        public string email { get; set; }
    }

    public class UltimoStatus
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ultimoStatusId { get; set; }
        public string uri { get; set; }
        public string nome { get; set; }
        public string siglaPartido { get; set; }
        public string uriPartido { get; set; }
        public string siglaUf { get; set; }
        public int idLegislatura { get; set; }
        public string urlFoto { get; set; }
        public string email { get; set; }
        public string data { get; set; }
        public string nomeEleitoral { get; set; }
        public Gabinete gabinete { get; set; }
        public string situacao { get; set; }
        public string condicaoEleitoral { get; set; }
        public string descricaoStatus { get; set; }
    }
}

File: app/src/shared/detailedDeputado.ts
export interface Gabinete {
    andar: string;
    email: string;
    nome: string;
    predio: string;
    sala: string;
    telefone: string;
}

export interface UltimoStatus {
    condicaoEleitoral: string;
    data: string;
    descricaoStatus: string;
    email: string;
    gabinete: Gabinete;
    id: number;
    idLegislatura: number;
    nome: string;
    nomeEleitoral: string;
    siglaPartido: string;
    siglaUf: string;
    situacao: string;
    uri: string;
    uriPartido: string;
    urlFoto: string;
}

export interface DetailedDeputado {
    cpf: string;
    dataFalecimento: string;
    dataNascimento: string;
    email: string;
    escolaridade: string;
    id: number;
    municipioNascimento: string;
    nomeCivil: string;
    redeSocial: string[];
    sexo: string;
    ufNascimento: string;
    ultimoStatus: UltimoStatus;
    uri: string;
    urlWebsite: string;
    despesas: Despesa[];
    frentes: Frente[];
    orgaos: Orgao[];
}

export interface Despesa {
    ano: string;
    cnpjCpfFornecedor: string;
    codDocumento: string;
    codLote: string;
    codTipoDocumento: string;
    dataDocumento: string;
    mes: string;
    nomeFornecedor: string;
    numDocumento: string;
    numRessarcimento: string;
    parcela: string;
    tipoDespesa: string;
    tipoDocumento: string;
    urlDocumento: string;
    valorDocumento: string;
    valorGlosa: string;
    valorLiquido: string;
}

export interface Frente {
    id: string;
    uri: string;
    titulo: string;
    idLegislatura: string;
}

export interface Orgao {
    idOrgao: string;
    uriOrgao: string;
    siglaOrgao: string;
    nomeOrgao: string;
    titulo: string;
    codTitulo: string;
    dataInicio: string;
    dataFim: string;
}